api/v1alpha1/datadogagent_types.go

```diff
@@ -42,15 +43,15 @@ type DatadogAgentSpec struct {
 	// The desired state of the Agent as an extended daemonset.
 	// Contains the Node Agent configuration and deployment strategy.
 	// +optional
-	Agent *DatadogAgentSpecAgentSpec `json:"agent,omitempty"`
+	Agent DatadogAgentSpecAgentSpec `json:"agent,omitempty"`
```

Comment:
I understand the general move to non-pointer to pointer, but why moving these ones from pointer to non-pointer?

Reply:
Great point - There are multiple reasons.
The goal was to have the high level feature fields (Feature being one on them, I hope it's not confusing) as non ptr structs so I could pass them down when building the defaulting instead of passing by reference the same struct and trying to get the diff of what was overridden (which would not work if the pointer is nil initially as we would work on a local copy).
Secondly, it helps distinguish if a section is disabled entirely (which will help for the next task of dealing with the Feature section to enable a lot of things depending on a small feature flag). So if the section is empty, we use the `Enabled` flag as false (and so is the internal structure) and can end up with an operator which does not have the agent for instance (which should be the case but is not possible at the moment IIRC).